crates/sui-rosetta/src/lib.rs

```diff
@@ -99,3 +111,40 @@ impl RosettaOfflineServer {
         axum::serve(listener, app).await.unwrap();
     }
 }
+
+#[derive(Clone)]
+pub struct CoinMetadataCache {
+    client: SuiClient,
+    metadata: Arc<Mutex<HashMap<TypeTag, Currency>>>,
```

Comment:
this cache will grow if we have a lot of different coin types, I am not sure if this will cause any problem but we might want to use `LruCache` instead to make sure we won't run out of memory.

Reply:
Leaving it like this for now to unblock Coinbase integration testing but created a ticket for that which we can take care of soon.
https://mysten.atlassian.net/browse/SEBP-97